### PR TITLE
[Ops][Performance] Warm up GDN prefill kernels before KV-cache allocation

### PR DESCRIPTION
**Pull request by prefersung**
### What this PR does / why we need it?

Upstream vLLM warms Qwen GDN prefill kernels during the V1 profile run. The current Ascend adapter replaces that method with a no-op, and its `_forward_core` returns when attention metadata is absent. As a result, the first real prefill remains the first execution of the Ascend GDN gating and chunked-prefill path.

This draft tests whether moving that first-use initialization into profiling improves cold-start behavior on Ascend A2/A3. It:

- intercepts the no-metadata GDN profile path and runs one 64-token prefill shape;
- warms `DeviceOperator.fused_gdn_gating` and the active Ascend GDN prefill backend;
- uses the fused CANN chunk operator when its runtime probe succeeds and PCP is disabled;
- otherwise warms the existing FLA/Triton JIT path;
- preserves the live recurrent-state layout for each backend (`[N,H,V,K]` for fused CANN and `[N,H,K,V]` for the FLA fallback);
- caches successful warmup signatures across GDN layers in the process;
- leaves failed signatures retryable and lets the normal first request initialize the backend;
- synchronizes before recording success and releases temporary cached NPU memory;
- loads only for `STANDARD_WORKER_PATCHES`, leaving the existing 310P no-op behavior unchanged.

The metadata-backed runtime path delegates unchanged to the existing Ascend `_forward_core` implementation.

This is deliberately framed as a hardware-validation experiment. The current Ascend FLA kernels are JIT-compiled Triton kernels; this PR does not claim an autotuning or OOM improvement until an A2/A3 cold-start A/B run demonstrates it.

### Does this PR introduce _any_ user-facing change?

No API or configuration change. It moves one internal GDN first-use execution into the existing V1 profile phase on supported standard-worker hardware.

### How was this patch tested?

Repository CI on the current head has passed:

- PR-title validation;
- Gitleaks;
- the complete pre-commit hook set, including Ruff check/format and repository policy checks;
- mypy.

Added CPU unit coverage for:

- FLA/Triton fallback dispatch;
- fused-CANN dispatch;
- backend-specific recurrent-state layouts;
- one warmup per process-level signature;
- retry after a failed warmup;
- profile-path interception;
- unchanged delegation for normal runtime metadata.

The focused pytest file has not yet run because maintainer-selected test CI has not been enabled for this draft.

### Hardware validation still required

Before marking the PR ready, an Ascend A2/A3 cold-start A/B run should record:

- service startup time to `/health=200`;
- profile-stage GDN warmup duration;
- first-request and second-request TTFT;
- first-request end-to-end latency;
- NPU memory before and after KV-cache allocation;
- the location of Triton JIT/backend initialization in the logs or profiler trace;
- deterministic output parity between the base and warmup branches.

The PR should be closed rather than promoted if the warmup only shifts latency into startup without improving first-request stability or total cold-start cost.

- vLLM Ascend base: `eb617eaf14d35d94fcbd391357d925d1833389a5`
- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/patch/worker/patch_gdn_prefill_warmup.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
+#
```
[the remark this answers is not in the document]
Comment (prefersung) [reply vllm_ascend/patch/worker/patch_gdn_prefill_warmup.py:1]: Updated the title to the repository-style `[Ops][Performance]` structure and kept the standard three-section PR description. I retained `Performance` rather than `Feature` because this draft changes initialization timing only and adds no new user-facing capability.

File: vllm_ascend/patch/worker/patch_gdn_prefill_warmup.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# mypy: ignore-errors
+
+import torch
```
[the remark this answers is not in the document]
Comment (prefersung) [reply vllm_ascend/patch/worker/patch_gdn_prefill_warmup.py:19]: Addressed in the latest commit by adding `import torch_npu  # noqa: F401` before using `torch.npu`.
